target/linux/ramips/image/mt7621.mk

```diff
@@ -9,6 +9,29 @@ DEFAULT_SOC := mt7621
 KERNEL_DTB += -d21
 DEVICE_VARS += ELECOM_HWNAME LINKSYS_HWNAME
 
+define Build/beeline-sb-flash-header
+	echo -ne "$$(echo 000001001c0000000000420000000000 | sed \
+		's/../\\x&/g')" | dd of=$@.tmp bs=1 count=16 \
+		conv=notrunc >/dev/null
+	dd if=$@ >> $@.tmp 2>/dev/null
+	mv $@.tmp $@
+	echo -n "hsqs" | dd of=$@ bs=1 seek=$$((0x41fff4)) count=4 \
+		conv=notrunc 2>/dev/null
+	echo -ne "$$(echo 5d436f740070ff00 | sed 's/../\\x&/g')" | dd \
+		of=$@.tmp bs=1 count=8 conv=notrunc 2>/dev/null
+	echo -ne "$$(dd if=$@ bs=16740340 count=1 2>/dev/null | gzip -c \
+		| tail -c 8 | od -An -tx4 -N4 --endian=big | tr -d ' \n' \
+		| awk '{ printf "%8s\n", $$0 }' | tr '0123456789abcdef' \
+		'fedcba9876543210' | sed 's/../\\x&/g')" | dd of=$@.tmp \
+		bs=1 seek=8 count=4 conv=notrunc 2>/dev/null
+	dd if=$@ >> $@.tmp 2>/dev/null
+	mv $@.tmp $@
+	echo -ne "$$(echo 0070ff00 | sed 's/../\\x&/g')" | dd of=$@ bs=1 \
+		seek=$$((0xff7004)) count=4 conv=notrunc 2>/dev/null
+	echo -n "HDR0" | dd of=$@ bs=1 seek=$$((0xff710c)) count=4 \
```

Comment:
`HDR0` is the deafult magic value for the Broadcom trx format, see https://github.com/openwrt/openwrt/blob/master/package/system/mtd/src/trx.c
If this is the same you could use the existing trx utile.

Reply:
Thanks for the idea. I have already looked at this tool. Unfortunately the formats are different. In this router, the magic is almost at the end of the image, **after** the rootfs data. Here is the signature from the OEM firmware at the screenshot. U-Boot is searching for HDR0 magic at every boot. The data on the left side of HDR0 is a signature. It isn't required for successful boot and necessary only for the web update from OEM firmware. The format is currently unknown.  
[OEM fw](https://static.beeline.ru/upload/images/sbflash10016.trx)
![image](https://user-images.githubusercontent.com/82600916/156997596-96096d79-5f50-4355-9ba6-054450c72538.png)

Reply:
@hauke, "brcm,trx" doesn't suite because we need padded to 4mb kernel. trx-nand too because we need 2 trx magic:
1st unusual TRX magic at the start of image: 0x746f435d
2nd TRX magic 'HDR0' at the end of image.
So, looks like it isn't possible to use one of the existing util untouched. What do you think?